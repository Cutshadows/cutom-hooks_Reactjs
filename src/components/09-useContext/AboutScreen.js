import React, {useContext} from 'react';
import UserContext from './userContext';


const AboutScreen = () => {
    const {user, setUser} = useContext(UserContext);
    const handleClick=()=>{
        setUser({});
    }
    return ( 
        <>
            <h1>ABOUT SCREEN</h1>
            <pre>
                {JSON.stringify(user, null, 3)}
            </pre>
            <button
                className="btn btn-warning"
                onClick={handleClick}>
                Logout
            </button>
        </>
     );
}
 
export default AboutScreen;