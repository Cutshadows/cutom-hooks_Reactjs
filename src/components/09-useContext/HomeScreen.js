import React, { useState, useContext } from 'react';
import UserContext from './userContext';


const HomeScreen = () => {
    const {user} = useContext(UserContext);

    return ( 
        <>
            <h1>HomeScreen</h1>

            <pre >
                {JSON.stringify(user, null, 3)}
            </pre>
        </>
     );
}
 
export default HomeScreen;