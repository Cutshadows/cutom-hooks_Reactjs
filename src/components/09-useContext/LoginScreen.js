import React,{useContext} from 'react';
import UserContext from './userContext';


const LoginScreen = () => {
    const {user, setUser} = useContext(UserContext);

    return ( 
        <>
            <h1>LoginScreen</h1>
            <hr/>
            <button
            className="btn btn-outline-success"
            onClick={()=>setUser({
                email:'main@email.com',
                name:'Monarca10'
            })}>
                Login
            </button>
        </>
     );
}
 
export default LoginScreen;