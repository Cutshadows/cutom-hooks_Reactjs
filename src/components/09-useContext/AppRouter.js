import React from 'react'

import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link,
    Redirect
} from 'react-router-dom';
import { NavBar } from './NavBar';
import AboutScreen from './AboutScreen';
import HomeScreen from './HomeScreen';
import LoginScreen from './LoginScreen';


export const AppRouter = () => {
    return (
        <Router>
            <div >
                <NavBar/>
                <div className="container">
                <Switch>
                    <Route exact path="/about"  component={AboutScreen} />
                    <Route exact path="/login" component={LoginScreen} />
                    <Route exact path="/" component={HomeScreen} />
                    <Redirect  to="/" />
                </Switch>
                </div>
            </div>
        </Router>
    )
}
