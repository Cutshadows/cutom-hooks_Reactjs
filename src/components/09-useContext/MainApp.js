import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import UserContext from './userContext';


const MainApp = () => {
    const [user, setUser] = useState({});

    return ( 
        <UserContext.Provider
            value={{
                user,
                setUser
           }}>

            <AppRouter/>
        </UserContext.Provider>
     );
}
 
export default MainApp;