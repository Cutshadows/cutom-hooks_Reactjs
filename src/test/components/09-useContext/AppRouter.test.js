import { mount } from 'enzyme';
import React from 'react';
import {AppRouter} from '../../../components/09-useContext/AppRouter';
import UserContext from '../../../components/09-useContext/userContext';

describe('Pruebas en AppRouter', () => {
    const user={
        id:1,
        name:'Fernando',
        email:'Fernandoq@email.com'
    }
    const wrapper=mount(
    <UserContext.Provider value={{
        user
    }}>
        <AppRouter />
    </UserContext.Provider>
    );
    test('debe de cargar las rutas', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})
