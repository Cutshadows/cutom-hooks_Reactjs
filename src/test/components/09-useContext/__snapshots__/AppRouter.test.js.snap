// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter debe de cargar las rutas 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      useContext
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    Array [
      <h1>
        HomeScreen
      </h1>,
      <pre>
        {
   "id": 1,
   "name": "Fernando",
   "email": "Fernandoq@email.com"
}
      </pre>,
    ]
  </div>
</div>
`;
